colcon_core/package_identification/python.py

```diff
@@ -100,5 +101,5 @@ def extract_dependencies(options):
             # remove environmental markers (separated by semicolons)
             # and version specifiers (separated by comparison operators)
             name = re.split(r';|<|>|<=|>=|==|!=', dep)[0].rstrip()
-            dependencies[dependency_type].add(name)
+            dependencies[dependency_type].add(DependencyDescriptor(name))
```

Comment:
The version constraints of the string should be extracted and added to the descriptor. This could be done in a separate patch if you prefer though.

Reply:
Yea, I'll do that in a separate commit.